services/brig/src/Brig/API/Client.hs

```diff
@@ -462,9 +475,17 @@ pubClient c =
       pubClientClass = clientClass c
     }
 
-legalHoldClientRequested :: UserId -> LegalHoldClientRequest -> (AppT r) ()
+legalHoldClientRequested ::
+  ( Member (Embed HttpClientIO) r,
+    Member NotificationSubsystem r,
+    Member Async r,
+    Member TinyLog r
+  ) =>
```

Comment:
will these `Async` constraints stay? I think it's a bit odd that we capture the semantics of this handler being asynchronous in the top level constraint. Ideally this should be abstracted away as especially access to `Async` is something that should be thoroughly tested, no?

Reply:
Good observation. We added another action to the effect which sends notifs async so we could get rid of all this noise.